auditbeat/magefile.go

```diff
@@ -93,191 +93,89 @@ func TestPackages() error {
-			args.Spec.ReplaceFile("/etc/{{.BeatName}}/{{.BeatName}}.reference.yml", referenceConfig)
-		case mage.Docker:
-			args.Spec.ExtraVar("user", "root")
-		default:
-			panic(errors.Errorf("unhandled package type: %v", pkgType))
-		}
-		if args.OS == "linux" {
-			rulesDest := archiveRulesDir
-			if pkgType != mage.TarGz {
-				rulesDest = linuxPkgRulesDir
-			}
-			args.Spec.Files[rulesDest] = sampleRules
-		}
-	}
+// IntegTest executes integration tests (it uses Docker to run the tests).
+func IntegTest() {
+	mage.AddIntegTestUsage()
+	defer mage.StopIntegTestEnv()
+	mg.SerialDeps(GoIntegTest, PythonIntegTest)
 }
 
-func generateReferenceConfig(spec mage.PackageSpec) error {
-	params := map[string]interface{}{
-		"Reference": true,
-		"ArchBits":  archBits,
-	}
-	return spec.ExpandFile(referenceConfigTemplate,
-		"{{.PackageDir}}/auditbeat.reference.yml", params)
+// UnitTest executes the unit tests.
+func UnitTest() {
+	mg.SerialDeps(GoUnitTest, PythonUnitTest)
 }
 
-func generateShortConfig(spec mage.PackageSpec) error {
-	params := map[string]interface{}{
-		"Reference": false,
-		"ArchBits":  archBits,
-	}
-	return spec.ExpandFile(shortConfigTemplate,
-		"{{.PackageDir}}/auditbeat.yml", params)
+// GoUnitTest executes the Go unit tests.
+// Use TEST_COVERAGE=true to enable code coverage profiling.
+// Use RACE_DETECTOR=true to enable the race detector.
+func GoUnitTest(ctx context.Context) error {
+	return mage.GoTest(ctx, mage.DefaultGoTestUnitArgs())
 }
 
-// archBits returns the number of bit width of the GOARCH architecture value.
-// This function is used by the auditd module configuration templates to
-// generate architecture specific audit rules.
-func archBits(goarch string) int {
-	switch goarch {
-	case "386", "arm":
-		return 32
-	default:
-		return 64
-	}
+// GoIntegTest executes the Go integration tests.
+// Use TEST_COVERAGE=true to enable code coverage profiling.
+// Use RACE_DETECTOR=true to enable the race detector.
+func GoIntegTest(ctx context.Context) error {
+	return mage.RunIntegTest("goIntegTest", func() error {
+		return mage.GoTest(ctx, mage.DefaultGoTestIntegrationArgs())
+	})
 }
 
-// Configs generates the auditbeat.yml and auditbeat.reference.yml config files.
-// Set DEV_OS and DEV_ARCH to change the target host for the generated configs.
-// Defaults to linux/amd64.
-func Configs() {
-	mg.Deps(makeConfigTemplates)
+// PythonUnitTest executes the python system tests.
+func PythonUnitTest() error {
+	mg.Deps(mage.BuildSystemTestBinary)
+	return mage.PythonNoseTest(mage.DefaultPythonTestUnitArgs())
+}
 
-	params := map[string]interface{}{
-		"GOOS":      mage.EnvOr("DEV_OS", "linux"),
-		"GOARCH":    mage.EnvOr("DEV_ARCH", "amd64"),
-		"ArchBits":  archBits,
-		"Reference": false,
+// PythonIntegTest executes the python system tests in the integration environment (Docker).
+func PythonIntegTest(ctx context.Context) error {
+	if !mage.IsInIntegTestEnv() {
+		mg.Deps(Fields)
```

Comment:
Should it also depend on `Dashboards`? Without it, I get the following error when running `mage integtest` in `beats/auditbeat`:


```
# mage integtest
[...]
>> python test: Integration Testing
E.......
======================================================================
ERROR: Test that the dashboards can be loaded with `setup --dashboards`
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/elastic/beats/auditbeat/tests/system/test_base.py", line 67, in test_dashboards
    shutil.copytree(kibana_dir, os.path.join(self.working_dir, "kibana"))
  File "/usr/lib/python2.7/shutil.py", line 171, in copytree
    names = os.listdir(src)
OSError: [Errno 2] No such file or directory: '/go/src/github.com/elastic/beats/auditbeat/build/kibana'
```

With `Dashboards`, I get further, but it still fails:

```
>> python test: Integration Testing
F.......
======================================================================
FAIL: Test that the dashboards can be loaded with `setup --dashboards`
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/elastic/beats/auditbeat/tests/system/test_base.py", line 80, in test_dashboards
    self.run_beat(extra_args=["setup", "--dashboards"], exit_code=0)
  File "/go/src/github.com/elastic/beats/auditbeat/tests/system/../../../metricbeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 164, in run_beat
    return proc.check_wait(exit_code)
  File "/go/src/github.com/elastic/beats/auditbeat/tests/system/../../../metricbeat/tests/system/../../../libbeat/tests/system/beat/beat.py", line 93, in check_wait
    exit_code, actual_exit_code)
AssertionError: Expected exit code to be 0, but it was 1
```

From `build/system-tests/run/test_base.Test.test_dashboards/auditbeat.log`:
```
2018-12-04T22:30:25.815Z        ERROR   instance/beat.go:795    Exiting: Failed to import index-pattern: Failed to load directory /go/src/github.com/elastic/beats/auditbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/6/index-pattern:
  error loading /go/src/github.com/elastic/beats/auditbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/6/index-pattern/auditbeat.json: blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];. Response: {"objects":[{"id":"auditbeat-*","type":"index-pattern","error":{"message":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}}]}
Exiting: Failed to import index-pattern: Failed to load directory /go/src/github.com/elastic/beats/auditbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/6/index-pattern:
  error loading /go/src/github.com/elastic/beats/auditbeat/build/system-tests/run/test_base.Test.test_dashboards/kibana/6/index-pattern/auditbeat.json: blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];. Response: {"objects":[{"id":"auditbeat-*","type":"index-pattern","error":{"message":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}}]}
```

I didn't look more into why this is happening. Does it happen for you, too?